src/sentry/processing/backpressure/redis.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any, Generator, Mapping, Sequence, Union
+
+import rb
+from redis import Redis
+from rediscluster import RedisCluster
+
+# Based on configuration, this could be:
+# - a `rb` (aka redis blaster) Cluster
+# - a `rediscluster` Cluster (actually `RetryingRedisCluster`)
+# - a straight `Redis` client (actually `FailoverRedis`)
+# - or any class configured via `client_class`.
+Cluster = Union[rb.Cluster, RedisCluster, Redis]
```

Comment:
taking a union like this feels like a smell -- it should be 3 separate apis for each of the types

Reply:
almost all of our redis-based services attempt to be agnostic across at least two of those types. I don't disagree that we could have better abstractions for those, but whether something points to a redis cluster or a single-node redis is runtime-configuration, therefore `sentry.utils.redis.cluster_manager.get` returns a union, and this code just inherits this design decision for the sake of consistency